### PR TITLE
Preverse error when return from failed precheck

### LINKED ISSUE #23: Timeout doesn't preserve original error

### PR DESCRIPTION
**Pull request by ammario**
Resolves #23

### CODE REVIEW

**Comment by ammario** [comment]
Waiting on @nhooyr to agree with these changes before merging.

**Comment by nhooyr** [comment]
Commented in #23

**Comment by ammario** [comment]
@nhooyr this is ready

File: retry.go [hunk 1]
```diff
@@ -237,18 +238,50 @@ func (r *Retry) Log(logFn func(error)) *Retry {
...
 // The retry must not be ran twice.
 func (r *Retry) Run(fn func() error) error {
+	var e Error
 	for {
-		err := r.preCheck()
-		if err != nil {
-			return err
+		e.Reason = r.preCheck()
+		if e.Reason != nil {
+			return e.nilIfEmpty()
```
Comment (nhooyr) [inline retry.go]: This can never be empty.

Comment (ammario) [reply retry.go]: I think it can... At least it was causing a test error because it was.

Comment (ammario) [reply retry.go]: Even if we can prove that is never empty, that function will catch mistakes in-case the behavior changes in the future.

**Pull request merged**